librz/core/rop.c

```diff
@@ -224,16 +224,15 @@ static bool rz_rop_print_json_mode(const RzCore *core, const RzCoreAsmHit *hit,
 	return true;
 }
 
-RZ_IPI void rz_core_rop_reg_info_free(RzRopRegInfo *reg_info) {
-	if (!reg_info) {
-		return;
-	}
+RZ_API void rz_core_rop_reg_info_free(RZ_NULLABLE RzRopRegInfo *reg_info) {
+	rz_return_if_fail(reg_info);
 	free(reg_info->name);
 	free(reg_info->value_transformations);
 	free(reg_info);
 }
 
-RZ_IPI RzRopRegInfo *rz_core_rop_reg_info_new(const RzCore *core, const RzILEvent *evt, const ut64 init_val, const ut64 new_val) {
+RZ_API RZ_NULLABLE RzRopRegInfo *rz_core_rop_reg_info_new(RZ_NONNULL const RzCore *core, RZ_NONNULL const RzILEvent *evt,
```

Comment:
You forgot ownership again. Please, try to focus on this, it's the same problem again and again. And remove `NULLABLE` from the return value

Reply:
The `_new` function can return NULL right? Should it be removed from the return val?